covariancematrix.go

```diff
@@ -80,3 +80,75 @@ func CovarianceMatrix(cov *mat64.Dense, x mat64.Matrix, wts []float64) *mat64.De
 	cov.Scale(1/(n-1), cov)
 	return cov
 }
+
+// CorrelationMatrix calculates a correlation matrix from a matrix of data,
+// using a two-pass algorithm. The matrix returned will be symmetric, square,
+// and positive-semidefinite.
```

Comment:
I know that in theory covariance matrices are symmetric and positive semidefinite, but are we sure that's true in finite precision world as well? Basically, we can guarantee that the matrix is symmetric and square, so we may just want to say that.